### PR TITLE
[DOCS] Data stream migration API

### PR DESCRIPTION
**Pull request by danhermann**
This is the final PR for #61046.

### CODE REVIEW

File: docs/reference/indices/migrate-to-data-stream.asciidoc [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+new data stream. The alias will be removed, and the new data stream will
+be named `<alias>`.
+
+Aliases must meet the following criteria to be migrated to a data stream:
+
+- Must have an index defined as the <<aliases-write-index,write index>>
+- All indices associated with the alias must have a mapping for the
+`@timestamp` field of type `date` or `date_nanos`
+- Must not have any <<filtered,filters>>
+- Must not have any <<aliases-routing,custom routing>>
```
Comment (jrodewig) [inline docs/reference/indices/migrate-to-data-stream.asciidoc:93]: I'd reorder this slightly so the required criteria comes first. I also added a sentence about the write index after conversion.
```suggestion
(Required, string)
Name of the index alias to convert to a data stream. The alias must meet the
following criteria:

- The alias has a <<aliases-write-index,write index>>.
- All indices for the alias have a `@timestamp` field mapping of a `date` or `date_nanos` field type.
- The alias has no <<filtered,filters>>.
- The alias does not use <<aliases-routing,custom routing>>.

{es} removes the alias and creates a data stream with the same name. The alias's
indices become hidden backing indices for the stream. The alias's write index
becomes the stream's write index.
```

**Review by jrodewig — APPROVED** [review]
LGTM. I made some suggestions, but feel free to ignore those if wanted. I'm good with shipping this as is. Thanks @danhermann!

**Comment by danhermann** [comment]
@elasticmachine update branch

**Review by martijnvg — APPROVED** [review]
LGTM

**Comment by jrodewig** [comment]
Went ahead and backported this with #66786 (7.x) and #66787 (7.11). That'll let me get backports of #66054 in as well.

**Pull request merged**